src/Rules/RuleErrorBuilder.php

```diff
@@ -182,8 +183,11 @@ public function addTip(string $tip): self
 	 * @phpstan-this-out self<T&TipRuleError>
 	 * @return self<T&TipRuleError>
 	 */
-	public function discoveringSymbolsTip(): self
+	public function discoveringSymbolsTip(?string $className = null): self
 	{
+		if (isset($className) && substr($className, 0, 7) === 'Drupal/') {
```

Comment:
A test would be nice, I'm pretty sure this code is wrong (It's `Drupal\`, not `Drupal/`) :) And please use `str_starts_with`.

The test should be written for one of the impacted rules, it doesn't matter which one. An error tip is tested with a third array item (https://phpstan.org/developing-extensions/rules#testing-the-rule) - besides the message and the line.

Reply:
You're right about the backslash. My mistake.

> And please use `str_starts_with`.

Hmm. I was under the impression that [PHPStan supports PHP >= 7.2](https://phpstan.org/user-guide/getting-started). The `str_starts_with()` function was only introduced in PHP 8.

I'll see what I can come up with for a test.